Adds Create separate CSS files per component section (#34)
emulsify-ds/documentation: details/webpack-and-build.md

There are also some other niceties for feedback \(progress plugin\) and cleanup \(to keep the `dist` directory clean\). Also, by default there is a separate webpack configuration for development \(`webpack/webpack.dev.js`\) and production \(`webpack/webpack.prod.js`\). The development file is used when running `yarn develop` and the production task is meant to be used by the `yarn build-webpack` command for deployments.

##### Create separate CSS files per component

To modify the default sass behavior to compile into individual files, take the following steps:

> 1. In `webpack/plugins.js`
> ```
> // Change this
> const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
>   filename: 'style.css',
>   chunkFilename: '[id].css',
> });
> 
> // To this
> const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
>   filename: "[name].css",
>   chunkFilename: "[id].css"
> });
> ```

Reviewer: ```suggestion

1. In `webpack/plugins.js` change

```
const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
  filename: 'style.css',
  chunkFilename: '[id].css',
});
```

To:

```
const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
  filename: '[name].css',
  chunkFilename: '[id].css'
});
```
```

> 2. In the webpack directory create a css directory with the following structure. Each .js imports their own .scss file.
> 
> * webpack
> * * css
> * * * main-css.js
> * * * custom-css.js

Reviewer: ```suggestion
2. In the webpack directory create a css directory with the following structure. (Each js file should import its own `<component>.scss` file. Look at `webpack/css.js` for an example.)

```
.
└──webpack
│  └──css
│     │   component-1.js
│     │   component-2.js
```

```

3. In webpack.common.js replace `entries.css` with

```
// CSS Files.

> glob.sync(`${webpackDir}/css/*js`).forEach((file) => {

Reviewer: I'm getting an error that `ReferenceError: webpackDir is not defined.` Is there a line missing here?

Reviewer: I wonder if there's a difference in what version of Emulsify you are using, and the latest in `master`

Reviewer 2: On line 6 I have the following for webpackDir
`const webpackDir = path.resolve(__dirname);`

> 3. In webpack.common.js replace `entries.css` with
> 
> ```
> // CSS Files.
> glob.sync(`${webpackDir}/css/*js`).forEach((file) => {
>   const baseFileName = path.basename(file);
>   const newfilePath = `css/${baseFileName.replace('.js', '')}`;
>   entries[newfilePath] = file;
> });
> ```

Reviewer: ```suggestion
3. In webpack.common.js replace

```
entries.css = path.resolve(webpackDir, 'css.js');
```

with

```
// CSS Files.
glob.sync(`${webpackDir}/css/*js`).forEach((file) => {
  const baseFileName = path.basename(file);
  const newfilePath = `css/${baseFileName.replace('.js', '')}`;
  entries[newfilePath] = file;
});
```
```

> 4. Now you can create your own Drupal libraries in the Drupal theme to import the files where you need them.

Reviewer: ```suggestion
4. Now you can define CSS files in your Drupal libraries only where you need them, just like you do with js.

5. Finally, Storybook only loads the main `styles.scss` file, by default. So make sure you add anything not globbed into that file to `.storybook/preview.js`.
```
